pkg/relay/ethereum/error_parse.go

```diff
@@ -22,14 +21,19 @@ func CreateErrorRepository(abiPaths []string) (ErrorRepository, error) {
 				return err
 			}
 			if !info.IsDir() && strings.HasSuffix(info.Name(), ".json") {
-				data, err := os.ReadFile(path)
+				f, err := os.Open(path)
 				if err != nil {
 					return err
 				}
+				defer f.Close()
 
-				contractABI, err := abi.JSON(bytes.NewReader(data))
-				for _, error := range contractABI.Errors {
-					errABIs = append(errABIs, error)
+				contractABI, err := abi.JSON(f)
+				if err != nil {
+					return err
```

Comment:
Thanks, sorry but it would be helpful if the file name/path is included  in the error @siburu

Reply:
Thank you! Fixed https://github.com/datachainlab/ethereum-ibc-relay-chain/pull/39/commits/aa347c27263ee5a4eeb5bcd6894ddaaa2e96e415

Reply:
Hm ... the same advice can be applied to the loop in `SendMsgs`.
I will fix it.

Reply:
fixed https://github.com/datachainlab/ethereum-ibc-relay-chain/pull/39/commits/f6fc68e2d4d398f3cd24662fd3fe4f15be45bdfc